{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# default_exp sumstat"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Sumstat module\n",
    "\n",
    "> read and extract Sumstat"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#hide\n",
    "from nbdev.showdoc import *"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#export\n",
    "import yaml\n",
    "import numpy as np\n",
    "import pandas as pd\n",
    "from scipy.stats import norm\n",
    "from cugg.utils import *"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#export\n",
    "def p2z(pval,beta,twoside=True):\n",
    "    if twoside:\n",
    "        pval = pval/2\n",
    "    z=np.abs(norm.ppf(pval))\n",
    "    ind=beta<0\n",
    "    z[ind]=-z[ind]\n",
    "    return z\n",
    "    \n",
    "class Sumstat:\n",
    "    def __init__(self,sumstat_path,config_file=None,rename=True):\n",
    "        self.ss = self.read_sumstat(sumstat_path,config_file,rename)\n",
    "        \n",
    "    def __repr__(self): \n",
    "        return \"sumstat:% s\" % (self.ss)\n",
    "        \n",
    "        #functions to read sumstats\n",
    "    def read_sumstat(self,file, config_file,rename):\n",
    "        if config_file is not None:\n",
    "            config_file = yaml.safe_load(open(config_file, 'r'))\n",
    "        return read_sumstat(file,config_file,rename)\n",
    "    \n",
    "    def extractbyregion(self,region):\n",
    "        sumstats = self.ss\n",
    "        idx = (sumstats.CHR == region[0]) & (sumstats.POS >= region[1]) & (sumstats.POS <= region[2])\n",
    "        print('this region',region,'has',sum(idx),'SNPs in Sumstat')\n",
    "        self.ss = sumstats[idx]\n",
    "        \n",
    "    def extractbyvariants(self,variants,notin=False):\n",
    "        idx = self.ss.SNP.isin(variants)\n",
    "        if notin:\n",
    "            idx = idx == False\n",
    "        #update sumstats\n",
    "        self.ss = self.ss[idx]\n",
    "        \n",
    "    def calculateZ(self):\n",
    "        self.ss['Z'] = list(p2z(self.ss.P,self.ss.BETA))\n",
    "        \n",
    "    def match_ss(self,bim):\n",
    "        self.ss = check_ss1(self.ss,bim)\n",
    "        \n",
    "        "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#export\n",
    "def read_sumstat(file, config,rename=True):\n",
    "    try:\n",
    "        sumstats = pd.read_csv(file, compression='gzip', header=0, sep='\\t', quotechar='\"')\n",
    "    except:\n",
    "        sumstats = pd.read_csv(file, header=0, sep='\\t', quotechar='\"')\n",
    "    if config is not None:\n",
    "        try:\n",
    "            ID = config.pop('ID').split(',')\n",
    "            sumstats = sumstats.loc[:,list(config.values())]\n",
    "            sumstats.columns = list(config.keys())\n",
    "            sumstats.index = namebyordA0_A1(sumstats[ID],cols=ID)\n",
    "        except:\n",
    "            raise ValueError(f'According to config_file, input summary statistics should have the following columns: %s' % list(config.values()))\n",
    "        sumstats.columns = list(config.keys())\n",
    "    if rename:\n",
    "        sumstats.SNP = 'chr'+sumstats.CHR.astype(str).str.strip(\"chr\") + ':' + sumstats.POS.astype(str) + '_' + sumstats.A0.astype(str) + '_' + sumstats.A1.astype(str)\n",
    "    sumstats.CHR = sumstats.CHR.astype(str).str.strip(\"chr\").astype(int)\n",
    "    sumstats.POS = sumstats.POS.astype(int)\n",
    "    if \"GENE\" in sumstats.columns.values():\n",
    "        sumstats.index = namebyordA0_A1(sumstats[[\"GENE\",\"CHR\",\"POS\",\"A0\",\"A1\"]],cols=[\"GENE\",\"CHR\",\"POS\",\"A0\",\"A1\"])\n",
    "    else:\n",
    "        sumstats.index = namebyordA0_A1(sumstats[[\"CHR\",\"POS\",\"A0\",\"A1\"]],cols=[\"CHR\",\"POS\",\"A0\",\"A1\"])\n",
    "    return sumstats"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Sumstat 2 vcf\n",
    "Following function take a sumstat pandas.df object that were processed and with correct headers ('CHR', 'POS', 'SNP', 'A0', 'A1',\"STAT\",\"SE\",\"P\"). and then change it into a vcf file with 1 samples. When the sumstat have gene name in it, the gene name will be saved to the info field."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#export\n",
    "def ss_2_vcf(ss_df,name = \"name\"):\n",
    "    ## Geno field\n",
    "    df = pd.DataFrame()\n",
    "    if \"SNP\" not in ss_df.columns:\n",
    "        ss_df['SNP'] = 'chr'+ss_df.CHR.astype(str).str.strip(\"chr\") + ':' + ss_df.POS.astype(str) + '_' + ss_df.A0.astype(str) + '_' + ss_df.A1.astype(str)\n",
    "    df[['#CHROM', 'POS', 'ID', 'REF', 'ALT']] = ss_df[['CHR', 'POS', 'SNP', 'A0', 'A1']].sort_values(['CHR', 'POS'])\n",
    "    ## Info field(Empty)\n",
    "    df['QUAL'] = \".\"\n",
    "    df['FILTER'] = \"PASS\"\n",
    "    df['INFO'] = \".\"\n",
    "    fix_header = [\"SNP\",\"A1\",\"A0\",\"POS\",\"CHR\",\"STAT\",\"SE\",\"P\"]\n",
    "    header_list = []\n",
    "    if \"GENE\" in ss_df.columns:\n",
    "        df['ID'] = ss_df['GENE'] + \":\" + ss_df['SNP']\n",
    "        df['INFO'] = \"GENE=\" + ss_df[\"GENE\"]\n",
    "        fix_header = [\"GENE\",\"SNP\",\"A1\",\"A0\",\"POS\",\"CHR\",\"STAT\",\"SE\",\"P\"]\n",
    "        header_list = ['##INFO=<ID=GENE,Number=1,Type=String,Description=\"The name of genes\">']\n",
    "    ### Fix headers\n",
    "    import time\n",
    "    header = '##fileformat=VCFv4.2\\n' + \\\n",
    "    '##FILTER=<ID=PASS,Description=\"All filters passed\">\\n' + \\\n",
    "    f'##fileDate={time.strftime(\"%Y%m%d\",time.localtime())}\\n'+ \\\n",
    "    '##FORMAT=<ID=STAT,Number=1,Type=Float,Description=\"Effect size estimate relative to the alternative allele\">\\n' + \\\n",
    "    '##FORMAT=<ID=SE,Number=1,Type=Float,Description=\"Standard error of effect size estimate\">\\n' + \\\n",
    "    '##FORMAT=<ID=P,Number=1,Type=Float,Description=\"The Pvalue corresponding to ES\">\\n' \n",
    "    ### Customized Field headers\n",
    "    for x in ss_df.columns:\n",
    "        if x not in fix_header:\n",
    "            Prefix = f'##FORMAT=<ID={x},Number=1,Type='\n",
    "            Type = str(type(ss_df[x][0])).replace(\"<class \\'\",\"\").replace(\"'>\",\"\").replace(\"numpy.\",\"\").replace(\"64\",\"\").capitalize().replace(\"Int\",\"Integer\")\n",
    "            Surfix = f',Description=\"Customized Field {x}\">'\n",
    "            header_list.append(Prefix+Type+Surfix)\n",
    "    ## format and sample field\n",
    "    df['FORMAT'] = \":\".join([\"STAT\",\"SE\",\"P\"] + ss_df.drop(fix_header,axis = 1).columns.values.tolist())\n",
    "    df[f'{name}'] = ss_df['STAT'].astype(str) + \":\" + ss_df['SE'].astype(str) + \":\" + ss_df['P'].astype(str) + \":\" + ss_df.drop(fix_header,axis = 1).astype(str).apply(\":\".join,axis = 1)\n",
    "    ## Rearrangment\n",
    "    df = df[['#CHROM', 'POS', 'ID', 'REF', 'ALT', 'QUAL', 'FILTER', 'INFO','FORMAT',f'{name}']]\n",
    "    # Add headers\n",
    "    header = header + \"\\n\".join(header_list) + \"\\n\"\n",
    "    return df,header"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Example input and output for ss_2_vcf()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 164,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>GENE</th>\n",
       "      <th>SNP</th>\n",
       "      <th>TSS_D</th>\n",
       "      <th>AF</th>\n",
       "      <th>MA_S</th>\n",
       "      <th>MA_F</th>\n",
       "      <th>P</th>\n",
       "      <th>STAT</th>\n",
       "      <th>SE</th>\n",
       "      <th>A1</th>\n",
       "      <th>A0</th>\n",
       "      <th>POS</th>\n",
       "      <th>CHR</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>DOC2B</td>\n",
       "      <td>chr17:150509_TA_T</td>\n",
       "      <td>-31141</td>\n",
       "      <td>0.185990</td>\n",
       "      <td>135</td>\n",
       "      <td>154</td>\n",
       "      <td>2.767300e-09</td>\n",
       "      <td>-0.353553</td>\n",
       "      <td>0.057822</td>\n",
       "      <td>T</td>\n",
       "      <td>TA</td>\n",
       "      <td>150509</td>\n",
       "      <td>chr17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>DOC2B</td>\n",
       "      <td>chr17:151035_C_T</td>\n",
       "      <td>-30615</td>\n",
       "      <td>0.185990</td>\n",
       "      <td>136</td>\n",
       "      <td>154</td>\n",
       "      <td>2.005370e-09</td>\n",
       "      <td>-0.360252</td>\n",
       "      <td>0.058372</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>151035</td>\n",
       "      <td>chr17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>DOC2B</td>\n",
       "      <td>chr17:151041_A_G</td>\n",
       "      <td>-30609</td>\n",
       "      <td>0.140097</td>\n",
       "      <td>102</td>\n",
       "      <td>116</td>\n",
       "      <td>8.754326e-03</td>\n",
       "      <td>-0.175506</td>\n",
       "      <td>0.066545</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>151041</td>\n",
       "      <td>chr17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>DOC2B</td>\n",
       "      <td>chr17:151596_G_A</td>\n",
       "      <td>-30054</td>\n",
       "      <td>0.096618</td>\n",
       "      <td>77</td>\n",
       "      <td>80</td>\n",
       "      <td>2.739677e-01</td>\n",
       "      <td>-0.088473</td>\n",
       "      <td>0.080737</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>151596</td>\n",
       "      <td>chr17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>DOC2B</td>\n",
       "      <td>chr17:151872_T_C</td>\n",
       "      <td>-29778</td>\n",
       "      <td>0.115942</td>\n",
       "      <td>92</td>\n",
       "      <td>96</td>\n",
       "      <td>9.243513e-02</td>\n",
       "      <td>0.132022</td>\n",
       "      <td>0.078229</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>151872</td>\n",
       "      <td>chr17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>995</th>\n",
       "      <td>DOC2B</td>\n",
       "      <td>chr17:281918_A_G</td>\n",
       "      <td>100268</td>\n",
       "      <td>0.335749</td>\n",
       "      <td>234</td>\n",
       "      <td>278</td>\n",
       "      <td>3.852610e-02</td>\n",
       "      <td>0.107802</td>\n",
       "      <td>0.051887</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>281918</td>\n",
       "      <td>chr17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>996</th>\n",
       "      <td>DOC2B</td>\n",
       "      <td>chr17:281965_T_G</td>\n",
       "      <td>100315</td>\n",
       "      <td>0.289855</td>\n",
       "      <td>206</td>\n",
       "      <td>240</td>\n",
       "      <td>8.156741e-03</td>\n",
       "      <td>0.140386</td>\n",
       "      <td>0.052740</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>281965</td>\n",
       "      <td>chr17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>997</th>\n",
       "      <td>DOC2B</td>\n",
       "      <td>chr17:282037_G_A</td>\n",
       "      <td>100387</td>\n",
       "      <td>0.010870</td>\n",
       "      <td>9</td>\n",
       "      <td>9</td>\n",
       "      <td>6.988573e-02</td>\n",
       "      <td>0.443654</td>\n",
       "      <td>0.243950</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>282037</td>\n",
       "      <td>chr17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>998</th>\n",
       "      <td>DOC2B</td>\n",
       "      <td>chr17:282127_G_A</td>\n",
       "      <td>100477</td>\n",
       "      <td>0.160628</td>\n",
       "      <td>122</td>\n",
       "      <td>133</td>\n",
       "      <td>8.626497e-01</td>\n",
       "      <td>0.011448</td>\n",
       "      <td>0.066118</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>282127</td>\n",
       "      <td>chr17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>999</th>\n",
       "      <td>DOC2B</td>\n",
       "      <td>chr17:282217_C_T</td>\n",
       "      <td>100567</td>\n",
       "      <td>0.083333</td>\n",
       "      <td>68</td>\n",
       "      <td>69</td>\n",
       "      <td>7.074454e-01</td>\n",
       "      <td>0.034193</td>\n",
       "      <td>0.091031</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>282217</td>\n",
       "      <td>chr17</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1000 rows × 13 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      GENE                SNP   TSS_D        AF  MA_S  MA_F             P  \\\n",
       "0    DOC2B  chr17:150509_TA_T  -31141  0.185990   135   154  2.767300e-09   \n",
       "1    DOC2B   chr17:151035_C_T  -30615  0.185990   136   154  2.005370e-09   \n",
       "2    DOC2B   chr17:151041_A_G  -30609  0.140097   102   116  8.754326e-03   \n",
       "3    DOC2B   chr17:151596_G_A  -30054  0.096618    77    80  2.739677e-01   \n",
       "4    DOC2B   chr17:151872_T_C  -29778  0.115942    92    96  9.243513e-02   \n",
       "..     ...                ...     ...       ...   ...   ...           ...   \n",
       "995  DOC2B   chr17:281918_A_G  100268  0.335749   234   278  3.852610e-02   \n",
       "996  DOC2B   chr17:281965_T_G  100315  0.289855   206   240  8.156741e-03   \n",
       "997  DOC2B   chr17:282037_G_A  100387  0.010870     9     9  6.988573e-02   \n",
       "998  DOC2B   chr17:282127_G_A  100477  0.160628   122   133  8.626497e-01   \n",
       "999  DOC2B   chr17:282217_C_T  100567  0.083333    68    69  7.074454e-01   \n",
       "\n",
       "         STAT        SE A1  A0     POS    CHR  \n",
       "0   -0.353553  0.057822  T  TA  150509  chr17  \n",
       "1   -0.360252  0.058372  T   C  151035  chr17  \n",
       "2   -0.175506  0.066545  G   A  151041  chr17  \n",
       "3   -0.088473  0.080737  A   G  151596  chr17  \n",
       "4    0.132022  0.078229  C   T  151872  chr17  \n",
       "..        ...       ... ..  ..     ...    ...  \n",
       "995  0.107802  0.051887  G   A  281918  chr17  \n",
       "996  0.140386  0.052740  G   T  281965  chr17  \n",
       "997  0.443654  0.243950  A   G  282037  chr17  \n",
       "998  0.011448  0.066118  A   G  282127  chr17  \n",
       "999  0.034193  0.091031  T   C  282217  chr17  \n",
       "\n",
       "[1000 rows x 13 columns]"
      ]
     },
     "execution_count": 164,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "test"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 163,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "##fileformat=VCFv4.2\n",
      "##FILTER=<ID=PASS,Description=\"All filters passed\">\n",
      "##fileDate=20220322\n",
      "##FORMAT=<ID=ES,Number=A,Type=Float,Description=\"Effect size estimate relative to the alternative allele\">\n",
      "##FORMAT=<ID=SE,Number=A,Type=Float,Description=\"Standard error of effect size estimate\">\n",
      "##FORMAT=<ID=P,Number=A,Type=Float,Description=\"The Pvalue corresponding to ES\">\n",
      "##FORMAT=<ID=TSS_D,Number=A,Type=Int,Description=\"Customized Field TSS_D\n",
      "##FORMAT=<ID=AF,Number=A,Type=Float,Description=\"Customized Field AF\n",
      "##FORMAT=<ID=MA_S,Number=A,Type=Int,Description=\"Customized Field MA_S\n",
      "##FORMAT=<ID=MA_F,Number=A,Type=Int,Description=\"Customized Field MA_F\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>FORMAT</th>\n",
       "      <th>name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr17</td>\n",
       "      <td>150509</td>\n",
       "      <td>chr17:150509_TA_T</td>\n",
       "      <td>TA</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>GENE = DOC2B</td>\n",
       "      <td>STAT:SE:P:TSS_D:AF:MA_S:MA_F</td>\n",
       "      <td>-0.35355327:0.05782158:2.7672995977620612e-09-...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr17</td>\n",
       "      <td>151035</td>\n",
       "      <td>chr17:151035_C_T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>GENE = DOC2B</td>\n",
       "      <td>STAT:SE:P:TSS_D:AF:MA_S:MA_F</td>\n",
       "      <td>-0.3602523:0.05837245:2.005370139560338e-09-30...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr17</td>\n",
       "      <td>151041</td>\n",
       "      <td>chr17:151041_A_G</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>GENE = DOC2B</td>\n",
       "      <td>STAT:SE:P:TSS_D:AF:MA_S:MA_F</td>\n",
       "      <td>-0.17550638:0.06654477:0.0087543256724295-3060...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr17</td>\n",
       "      <td>151596</td>\n",
       "      <td>chr17:151596_G_A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>GENE = DOC2B</td>\n",
       "      <td>STAT:SE:P:TSS_D:AF:MA_S:MA_F</td>\n",
       "      <td>-0.08847341:0.08073738:0.2739676780558129-3005...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr17</td>\n",
       "      <td>151872</td>\n",
       "      <td>chr17:151872_T_C</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>GENE = DOC2B</td>\n",
       "      <td>STAT:SE:P:TSS_D:AF:MA_S:MA_F</td>\n",
       "      <td>0.13202243:0.07822862:0.0924351300538036-29778...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>995</th>\n",
       "      <td>chr17</td>\n",
       "      <td>281918</td>\n",
       "      <td>chr17:281918_A_G</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>GENE = DOC2B</td>\n",
       "      <td>STAT:SE:P:TSS_D:AF:MA_S:MA_F</td>\n",
       "      <td>0.107802205:0.051887173:0.03852610270631691002...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>996</th>\n",
       "      <td>chr17</td>\n",
       "      <td>281965</td>\n",
       "      <td>chr17:281965_T_G</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>GENE = DOC2B</td>\n",
       "      <td>STAT:SE:P:TSS_D:AF:MA_S:MA_F</td>\n",
       "      <td>0.14038579:0.052740265:0.008156741448439610031...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>997</th>\n",
       "      <td>chr17</td>\n",
       "      <td>282037</td>\n",
       "      <td>chr17:282037_G_A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>GENE = DOC2B</td>\n",
       "      <td>STAT:SE:P:TSS_D:AF:MA_S:MA_F</td>\n",
       "      <td>0.4436538:0.24394979:0.0698857298190166100387:...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>998</th>\n",
       "      <td>chr17</td>\n",
       "      <td>282127</td>\n",
       "      <td>chr17:282127_G_A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>GENE = DOC2B</td>\n",
       "      <td>STAT:SE:P:TSS_D:AF:MA_S:MA_F</td>\n",
       "      <td>0.011447639:0.066118255:0.86264972708922861004...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>999</th>\n",
       "      <td>chr17</td>\n",
       "      <td>282217</td>\n",
       "      <td>chr17:282217_C_T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>GENE = DOC2B</td>\n",
       "      <td>STAT:SE:P:TSS_D:AF:MA_S:MA_F</td>\n",
       "      <td>0.034192912:0.091031134:0.70744535873019510056...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1000 rows × 10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    #CHROM     POS                 ID REF ALT QUAL FILTER          INFO  \\\n",
       "0    chr17  150509  chr17:150509_TA_T  TA   T    .   PASS  GENE = DOC2B   \n",
       "1    chr17  151035   chr17:151035_C_T   C   T    .   PASS  GENE = DOC2B   \n",
       "2    chr17  151041   chr17:151041_A_G   A   G    .   PASS  GENE = DOC2B   \n",
       "3    chr17  151596   chr17:151596_G_A   G   A    .   PASS  GENE = DOC2B   \n",
       "4    chr17  151872   chr17:151872_T_C   T   C    .   PASS  GENE = DOC2B   \n",
       "..     ...     ...                ...  ..  ..  ...    ...           ...   \n",
       "995  chr17  281918   chr17:281918_A_G   A   G    .   PASS  GENE = DOC2B   \n",
       "996  chr17  281965   chr17:281965_T_G   T   G    .   PASS  GENE = DOC2B   \n",
       "997  chr17  282037   chr17:282037_G_A   G   A    .   PASS  GENE = DOC2B   \n",
       "998  chr17  282127   chr17:282127_G_A   G   A    .   PASS  GENE = DOC2B   \n",
       "999  chr17  282217   chr17:282217_C_T   C   T    .   PASS  GENE = DOC2B   \n",
       "\n",
       "                           FORMAT  \\\n",
       "0    STAT:SE:P:TSS_D:AF:MA_S:MA_F   \n",
       "1    STAT:SE:P:TSS_D:AF:MA_S:MA_F   \n",
       "2    STAT:SE:P:TSS_D:AF:MA_S:MA_F   \n",
       "3    STAT:SE:P:TSS_D:AF:MA_S:MA_F   \n",
       "4    STAT:SE:P:TSS_D:AF:MA_S:MA_F   \n",
       "..                            ...   \n",
       "995  STAT:SE:P:TSS_D:AF:MA_S:MA_F   \n",
       "996  STAT:SE:P:TSS_D:AF:MA_S:MA_F   \n",
       "997  STAT:SE:P:TSS_D:AF:MA_S:MA_F   \n",
       "998  STAT:SE:P:TSS_D:AF:MA_S:MA_F   \n",
       "999  STAT:SE:P:TSS_D:AF:MA_S:MA_F   \n",
       "\n",
       "                                                  name  \n",
       "0    -0.35355327:0.05782158:2.7672995977620612e-09-...  \n",
       "1    -0.3602523:0.05837245:2.005370139560338e-09-30...  \n",
       "2    -0.17550638:0.06654477:0.0087543256724295-3060...  \n",
       "3    -0.08847341:0.08073738:0.2739676780558129-3005...  \n",
       "4    0.13202243:0.07822862:0.0924351300538036-29778...  \n",
       "..                                                 ...  \n",
       "995  0.107802205:0.051887173:0.03852610270631691002...  \n",
       "996  0.14038579:0.052740265:0.008156741448439610031...  \n",
       "997  0.4436538:0.24394979:0.0698857298190166100387:...  \n",
       "998  0.011447639:0.066118255:0.86264972708922861004...  \n",
       "999  0.034192912:0.091031134:0.70744535873019510056...  \n",
       "\n",
       "[1000 rows x 10 columns]"
      ]
     },
     "execution_count": 163,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "print(ss_2_vcf(test)[1])\n",
    "ss_2_vcf(test)[0]"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
